src/Cookie.php

```diff
@@ -40,88 +71,120 @@ final class Cookie
     private string $value;
 
     /**
-     * @var string|null domain of the cookie
+     * @var DateTimeInterface|null RFC-1123 date at which the cookie expires.
+     * @see https://tools.ietf.org/html/rfc6265#section-4.1.1
      */
-    private ?string $domain = null;
+    private ?DateTimeInterface $expire = null;
 
     /**
-     * @var int the timestamp at which the cookie expires. This is the server timestamp.
-     * Defaults to 0, meaning "until the browser is closed".
+     * @var int|null maximum age of cookie in seconds
      */
-    private int $expire = 0;
+    private ?int $maxAge = null;
```

Comment:
I don't think we need to store both maxAge and expire. One could be obtained from another on header generation.

As for headers it's better to always send both max age and expires (for HTTP 1.0 clients) that result in the same expiration time.

Reply:
Yep, I agree. But I see one problem here. A valid cookie can have different expiry and maxAge. And according to RFC the maxAge will have precedence in that case. If we don't store maxAge and generate from expiry.
What about cookies that are not set by yii backend, cookies that we just read with `Cookie::fromSetCookieString` if we don't store maxAge and expiry separately, and cookie has both set on client side, we would just change that cookie with in an unexpected way for the user. What if some frontend or implementation somehow depends on it. I'm sure that such behavior (setting different expiry and maxAge) is wrong at all, but it is valid per RFC, and it is the main reason I'm storing maxAge.

Reply:
Technically it can have different expiry and maxAge but it doesn't make sense to set it like that.

> What about cookies that are not set by yii backend, cookies that we just read

We can behave as RFC is suggesting: maxAge will have precedence.

Reply:
@samdark It can make sense in case if your custom client implementation somehow depends on that (not sure why).
So you propose just to set `Max-Age` anytime the expiry is set? And to also set `Expiry` to the value of  `Max-Age` when parsing the cookie from response and only `Max-Age` is there?

Reply:
Also take into account that `withMaxAge` method is still useful, as it allows to just set expiry in seconds instead of creating `DateTimeInterface`, it can be renamed to `validFor` as it was before, but we will not have a way to set `Max-Age` explicitly if we remove `$maxAge` property.

Reply:
I don't think there's a need to make a difference between these two so:

1. Keep one value as private property.
2. Keep two methods of setting it.
3. Add both headers to response.
4. Parse incoming request according to RFC i.e. if Max-Age present, use it. Otherwise, use Expiry.

Reply:
My current `maxAge` (will rename it to `withMaxAge`) method receives `DateInterval` type argument, I still would like to have a method that can also accept integer argument (seconds until expiry) as it can be handy sometimes, but I don't want to add another method for that only and also I see a use for `DateInterval` either (as it is now). The union types would (type hinting) would solve that problem, but unfortunately that RFC will land in PHP 8 by the end of year only.

Do you think we need another method (if we keep `withMaxAge` at all), or keep just this signature until union types and will update (BC break) in some future version?

Reply:
or make `withMaxAge` receive integer instead of `DateInterval` for now to be consistent `Max-Age` cookie attribute that is integer?

P.S. sorry for comment spamming.

Reply:
Interval is enough, I think. It is convenient to use it even with seconds:

```php
$seconds = 7200;

$interval = new DateInterval('PT' . $seconds . 'S');
// or
$interval = DateInterval::createFromDateString($seconds . ' seconds');
```